src/inference/src/cpp_interfaces/interface/ie_iinfer_request_internal.cpp

```diff
@@ -157,6 +159,175 @@ void IInferRequestInternal::SetBlob(const std::string& name, const Blob::Ptr& us
+                        param->get_partial_shape(),
+                        "batch ",
+                        batch,
+                        "doesn't match with total blobs count: ",
+                        input_size);
+    }
+    OPENVINO_ASSERT(batch_idx == 0,
+                    "set_input_tensors/set_tensors is not currently supported for batch dimension index ",
+                    batch_idx,
+                    " != 0");
+
+    // In future consider checking if blobs point to contiguous range of memory and use single 'SetBlob' instead
+    auto tmp_desc = blobs[0]->getTensorDesc();
+    tmp_desc.getDims()[batch_idx] = input_size;
+    auto blockingDims = tmp_desc.getBlockingDesc().getBlockDims();
+    blockingDims[batch_idx] = input_size;
+    auto blockingDesc = BlockingDesc(blockingDims, tmp_desc.getBlockingDesc().getOrder());
+    auto batched_desc = InferenceEngine::TensorDesc(tmp_desc.getPrecision(), tmp_desc.getDims(), blockingDesc);
+    auto desc_to_string = [](const TensorDesc& desc) {
+        std::stringstream s;
+        s << "{ " << desc.getLayout() << " " << desc.getPrecision().name();
+        s << "dim=(";
+        for (const auto& d : desc.getDims()) {
+            s << " " << d;
+        }
+        s << " ) }";
+        return s.str();
+    };
+    std::for_each(blobs.begin(), blobs.end(), [&batched_desc, &batch_idx, &desc_to_string](const Blob::Ptr& item) {
+        auto item_desc = item->getTensorDesc();
+        item_desc.getDims()[batch_idx] = batched_desc.getDims()[batch_idx];
+        OPENVINO_ASSERT(item_desc.getDims() == batched_desc.getDims() &&
+                            item_desc.getLayout() == batched_desc.getLayout() &&
+                            item_desc.getPrecision() == batched_desc.getPrecision() &&
+                            item_desc.getBlockingDesc().getOrder() == batched_desc.getBlockingDesc().getOrder(),
+                        "set_input_tensors/set_tensors error. Blob ",
+                        desc_to_string(item_desc),
+                        " is not compatible with batched blob ",
+                        desc_to_string(batched_desc));
+    });
+}
+
+void IInferRequestInternal::convertBatchedBlob(const std::string& name, const BatchedBlob::Ptr& batched_blob) {
+    auto tmp_desc = batched_blob->getBlob(0)->getTensorDesc();
+    tmp_desc.getDims()[0] = batched_blob->size();
+    auto blockingDims = tmp_desc.getBlockingDesc().getBlockDims();
+    blockingDims[0] = batched_blob->size();
+    auto blockingDesc = BlockingDesc(blockingDims, tmp_desc.getBlockingDesc().getOrder());
+    auto batched_desc = InferenceEngine::TensorDesc(tmp_desc.getPrecision(), tmp_desc.getDims(), blockingDesc);
+    std::shared_ptr<RemoteContext> remote_context;
+    MemoryBlob::Ptr mem_blob;
+    try {
+        auto net = getPointerToExecutableNetworkInternal();
+        if (net) {
+            remote_context = net->GetContext();
+        }
+    } catch (...) {
```

Comment:
Catch only `NotImplemented`. other exceptions should be rethrown.

Reply:
Agree